Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -35,6 +35,10 @@ public struct Tensor<Scalar: TensorFlowScalar> {
   /// - Note: `handle` is public to allow user defined ops, but should not normally be used.
   public let handle: TensorHandle<Scalar>
 
+  /// An internal marker to identify scalar zero tensors, for use in optimizations.
```

Comment:
Could this live on the TensorHandle?

Reply:
Sorry, kept meaning to check on this and got distracted. Would there be an advantage to having it on TensorHandle? I located it on Tensor because that's the point at which it impacts calculations. Only Tensor addition and subtraction are selectively converted to no-ops, so I didn't see a need to place this deeper within TensorHandle. The logic for marking zeroes would also need to be placed in Tensor.zero, so it made sense to me to have this all at the same level.

Reply:
This might fix your crasher as well. Also, another tensor type might handle zeros itself internally. In addition it makes the Tensor type bigger which makes it slower to copy around. We could also make this a completely different TensorHandle subclass.